api/admin/controller.py

```diff
@@ -805,6 +807,81 @@ def genres(self):
             })
         return data
 
+class CustomListsController(CirculationManagerController):
+    def custom_lists(self):
+        library = flask.request.library
+
+        data_source = DataSource.lookup(self._db, DataSource.LIBRARY_STAFF)
+
+        if flask.request.method == "GET":
+            custom_lists = []
+            for list in library.custom_lists:
+                entries = []
+                for entry in list.entries:
+                    entries.append(dict(id=entry.edition.primary_identifier.urn,
```

Comment:
Is there something better I could be using to identify editions? It needs to be something the client can get from the opds entry.

Reply:
I think it would be better to identify the list items by their work (through permanent work ID). CustomListEntry.editions is used for situations like the NYT best-sellers, where we know that a book with certain bibliographic information is on a list, but we don't know if we have any licenses for that book and we're not convinced that the bibliographic information is even accurate. When you're editing a list by choosing books from search results, you know that the books are already in the system as Works.

It's currently possible for multiple Works to have the same permanent work ID, but the same is true of an Edition's primary identifier, so this change shouldn't cause a new problem.